### PR TITLE
Add a main_input_name attribute to all models

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR adds a new attribute to all models called `main_input_name` which allows us to know whether the model expects `input_ids`, `pixel_values`, `input_values` or `input_features` as a first argument. A test is also added to check the correct value is set for all models, by inspecting the signature.

We could also add some magic property to do what's tested, but I feel an explicit attribute is more customizable for future use cases.

### CODE REVIEW

File: src/transformers/models/imagegpt/modeling_imagegpt.py [hunk 1]
```diff
@@ -495,6 +495,7 @@ class ImageGPTPreTrainedModel(PreTrainedModel):
     config_class = ImageGPTConfig
     load_tf_weights = load_tf_weights_in_imagegpt
     base_model_prefix = "transformer"
+    main_input_name = "pixel_values"
```
Comment (patrickvonplaten) [inline src/transformers/models/imagegpt/modeling_imagegpt.py:498]:
```suggestion
    main_input_name = "input_ids"
```

File: tests/test_modeling_common.py [hunk 2]
```diff
@@ -1315,6 +1315,13 @@ def test_model_common_attributes(self):
             x = model.get_output_embeddings()
             self.assertTrue(x is None or isinstance(x, nn.Linear))
 
+    def test_model_main_input_name(self):
```
Comment (patrickvonplaten) [inline tests/test_modeling_common.py:1318]: Very nice test!

File: src/transformers/models/imagegpt/modeling_imagegpt.py (hunk 1 above)
Comment (patrickvonplaten) [reply src/transformers/models/imagegpt/modeling_imagegpt.py:498]: About to merge the other PR

**Review by patrickvonplaten — APPROVED** [review]
Thanks for working on this. It'll make generate a bit cleaner as well :-)

**Review by LysandreJik — APPROVED** [review]
Love it, this is great and very explicit.

**Pull request merged**